Sources/Sentry/Profiling/SentryContinuousProfiler.mm

```diff
@@ -159,14 +164,32 @@ + (void)scheduleTimer
 
 + (void)timerExpired
 {
-    std::lock_guard<std::mutex> l(_threadUnsafe_gContinuousProfilerLock);
-    if (![_threadUnsafe_gContinuousCurrentProfiler isRunning]) {
-        SENTRY_LOG_WARN(@"Current profiler is not running. Sending whatever data it has left "
-                        @"and disabling the timer from running again.");
-        disableTimer();
+    {
+        std::lock_guard<std::mutex> l(_threadUnsafe_gContinuousProfilerLock);
+        if (![_threadUnsafe_gContinuousCurrentProfiler isRunning]) {
+            SENTRY_LOG_WARN(@"Current profiler is not running. Sending whatever data it has left "
+                            @"and disabling the timer from running again.");
+            disableTimer();
+        }
+
+        _sentry_threadUnsafe_transmitChunkEnvelope();
+
+        if (!_stopCalled) {
+            return;
+        }
     }
 
-    _sentry_threadUnsafe_transmitChunkEnvelope();
+    [self stopTimerAndCleanup];
+}
+
++ (void)stopTimerAndCleanup
+{
+    std::lock_guard<std::mutex> l(_threadUnsafe_gContinuousProfilerLock);
```

Comment:
`h`: Is it okay to leave the lock here? Shouldn't we instead move the code of `stopTimerAndCleanup` to `imerExpired` and keep it in the lock?

Reply:
I factored this method out to make stopping the continuous profiler cleaner in clearTestState, and didn't want that to cause any threading issues, so factored it out in a way that this will acquire the lock separately from the one acquired in the normal codepath that calls stopTimerAndCleanup in SentryContinuousProfiler.stop